src/main/java/org/opentripplanner/middleware/docs/PublicApiDocGenerator.java

```diff
@@ -0,0 +1,416 @@
+package org.opentripplanner.middleware.docs;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.opentripplanner.middleware.OtpMiddlewareMain;
+import org.opentripplanner.middleware.utils.YamlUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Predicate;
+
+import static org.opentripplanner.middleware.utils.ConfigUtils.getConfigPropertyAsText;
+
+/**
+ * Class that generates the public-facing API documentation.
+ */
+public class PublicApiDocGenerator {
+    private static final Logger LOG = LoggerFactory.getLogger(PublicApiDocGenerator.class);
+    private static final HttpClient client = HttpClient.newBuilder().build();
+    private static final String API_TEMPLATE_FILE = "api-swagger-template.yaml";
+
+    /**
+     * List of public API paths published in the public-facing API documentation.
+     * TODO: List any new public endpoints here in order for them to be published.
+     */
+    private static final String[] PUBLIC_PATHS = new String[] {
+        "api/secure/monitoredtrip",
+        "api/secure/triprequests",
+        "api/secure/user",
+        "otp",
+        "pelias"
+    };
+    private static final String DEFINITIONS_REF = "#/definitions/";
+    private static final String REF_FIELD = "$ref";
+    private static final String ABSTRACT_USER = "AbstractUser";
+    private static final String OTP_USER = "OtpUser";
+    private static final String AUTOGENERATED_DOC_URL = "http://localhost:4567/doc.yaml";
+    // Use the system temporary directory to write the output file
+    private static final String OUTPUT_FILE = System.getProperty("java.io.tmpdir") + File.separator + "otp-middleware-swagger.yaml";
+    private static final String JAR_PREFIX = "otp-middleware-";
+
+    private static final String AWS_API_SERVER = getConfigPropertyAsText("AWS_API_SERVER");
+    private static final String AWS_API_STAGE = getConfigPropertyAsText("AWS_API_STAGE");
+
+    // Cached variables.
+    private final ObjectNode rootNode;
+    private final JsonNode templateJson;
+    private final ObjectNode pathsNode;
+    private final ObjectNode definitionsNode;
+    private final ArrayNode tagsNode;
```

Comment:
These variables don't need the `Node` suffix. That's like calling an array of strings `stringsArray` instead of `strings`. Recommend changing:
- rootNode -> swaggerRoot
- templateJson -> supplements
- pathsNode -> paths
- defintionsNode -> defitinitions
- tagsNode -> tags

Reply:
Addressed in https://github.com/ibi-group/otp-middleware/pull/59/commits/4e3253e4b58640c8a42db4a24d02ac2c3319a3a7